### DISCUSSION TITLE
New Level: Selfdestruct and proxies

### DISCUSSION THREAD

**Discussion by gonzalpetraglia** [OpenZeppelin/ethernaut#514] [Level idea]
Hi,

I was thinking about a new level for the game. In this level the player should rediscover the multisig parity issue. I think this is important to be known because of the explosion of new developers coming to Solidity/Ethereum and because of the rapid growth on the usage of proxy related patterns(for example ZOS upgradeability system, [some storage layout ](https://medium.com/1milliondevs/solidity-storage-layout-for-proxy-contracts-and-diamonds-c4f009b6903), and many others). Though some of the previously mentioned tools have checks to avoid this type of issue, I still think this is worth playing with and knowing it in detail.

What do you think?

**Comment by abcoathup** [comment]
Hi @gonzalpetraglia!  Thanks for the suggested level, it is really appreciated.

OpenZeppelin Upgrades Plugins check for upgrade safety, including checking for `selfdestruct`.

Please wait until there has been discussion on this suggested level before writing any code or submitting a Pull Request.  We don’t want you to waste your time!

**Comment by gonzalpetraglia** [comment]
I'm thinking about other levels similar to this one regarding enums or struct modification during an update.

For example a level, where a user could change a value in a struct before triggering an update that modifies said struct such that the user is able to drain the contract.

What do you think?
